### PR TITLE
Add tests and travis config.

### PR DESCRIPTION
**Pull request by SurferJeffAtGoogle**

### CODE REVIEW

File: api_test.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                reply = self.urlopen(next_link['target'], payload)
+            else:
+                break
+        if reply['status'] == 'success':
+            return result
+        else:
+            raise Failure(reply['status'], errors)
+
+
+with BrowserState(HOST) as state:
```
Comment (tmatsuo) [inline api_test.py:115]: Can you use `unittest` module?
https://docs.python.org/2/library/unittest.html

File: api_test.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    def handle_response(self, response):
+        """Expects a successful HTTP response.
+
+        :arg response: a return value from urllib2.urlopen()
+        :returns a decoded json dict.
+        """
+        assert response.getcode() >= 200
+        assert response.getcode() < 300
+        reply = json.load(response)
+        pprint.pprint(reply)
```
Comment (tmatsuo) [inline api_test.py:67]: Is `pprint.pprint(reply)` leftover for debugging?

File: api_test.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+    def urlopen(self, target, keyval):
+        """Call urllib2.urlopen() for the target link.
+
+        :arg target: a url or partial url relative to host name.
+        :returns a decoded json dict.
+        """
+        url = urlparse.urljoin(self.host, target)
+        print url
+        pprint.pprint(keyval)
```
Comment (tmatsuo) [inline api_test.py:80]: ditto

**Comment by tmatsuo** [comment]
Tests are always good :)

I put some comments. Also let's squash the commits before merging.

File: api_test.py (hunk 1 above)
Comment (SurferJeffAtGoogle) [inline api_test.py:115]: Done.

File: api_test.py (hunk 2 above)
Comment (SurferJeffAtGoogle) [inline api_test.py:67]: I'm used to somewhat verbose tests.  If the test fails, I want to have an idea of why it failed and how to fix it, without having to run the test myself.  Is that different from our usual pattern?

**Comment by SurferJeffAtGoogle** [comment]
I couldn't figure out how to squash in this branch, so I created a new branch.  New pull request coming soon.

**Pull request closed without merging**